File: backend-app/src/main/java/com/backend_app/backend_app/service/RolService.java
package com.backend_app.backend_app.service;

import com.backend_app.backend_app.domain.Rol;
import com.backend_app.backend_app.dto.RolDTO;
import org.springframework.data.domain.Page;

import java.util.List;
import java.util.Optional;

public interface RolService {

    List<RolDTO> listadoRoles();

    Optional<RolDTO> buscarRolPorId(long idRol);

    RolDTO convertirADTO(Rol rol);

    void insertarRol(RolDTO rolDTO);

    void actualizarRol(RolDTO rolDTO);

    void eliminarRol(RolDTO rolDTO);

    List<RolDTO> listadoRolesCompletos();



}

File: frontend-app/src/app/services/rol.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Observable } from "rxjs";
import { Rol } from "../domain/Rol";

@Injectable({
    providedIn: 'root'
})
export class RolService{

    private url:string = 'http://localhost:8080/api/roles';

    constructor(private http: HttpClient){

    }

    listadoRoles(): Observable<Rol[]>{
        return this.http.get<Rol[]>(this.url);
    }
    
        
    buscarRolPorId(id: number): Observable<Rol>{
        return this.http.get<Rol>(`${this.url}/${id}`);
    }
    
    guardarRol(rol: Rol){
        return this.http.post('http://localhost:8080/api/roles/guardar', rol);
    }
    
    editarRol(rol: Rol) {
        return this.http.put(`${this.url}/actualizar`, rol);
    }
    
    eliminarRol(idRol: number){
        return this.http.delete(`${this.url}/${idRol}`);
    }

    listadoRolPaginacion(page: number): Observable<any>{
        return this.http.get<any[]>(`${this.url}/page/${page}`);
    }   

    listadoRolesCompletos(): Observable<Rol[]>{
        return this.http.get<Rol[]>(`${this.url}/completos`);
    }

}